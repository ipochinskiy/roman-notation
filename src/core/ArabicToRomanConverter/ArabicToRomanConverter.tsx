import React, { useState } from 'react';

import { Card, Input, Text } from '../../components';

import styles from './ArabicToRomanConverter.module.scss';

export const ArabicToRomanConverter: React.FC = () => {
  const [romanValue, setRomanValue] = useState('42');

  const onChangeHandler = (newValue: string) => {
    setRomanValue(newValue);
  };

  return (
    <div className={styles.converter}>
      <Card>
        <Input placeholder='42' type='number' value={romanValue} fullWidth onChange={onChangeHandler} />
        {romanValue && (
          <div className={styles.info}>
            <Text color='tertiary' size='small'>
              Roman notation:
            </Text>
            <Text color='tertiary' size='large'>
              {romanValue}
            </Text>
          </div>
        )}
      </Card>
    </div>
  );
};
