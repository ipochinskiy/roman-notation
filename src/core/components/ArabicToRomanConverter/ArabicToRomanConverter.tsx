import React from 'react';

import { Card, Input, Text } from '../../../components';
import { useMemoizedCallback } from '../../hooks/use-memoized-callback';
import { mapToRoman } from '../../utils/map-to-roman';

import styles from './ArabicToRomanConverter.module.scss';

export const ArabicToRomanConverter: React.FC = () => {
  const { value, error, doMap } = useMemoizedCallback(mapToRoman);

  return (
    <Card>
      <Input
        data-testid='arabic-input'
        min={1}
        minLength={1}
        placeholder='42'
        type='number'
        fullWidth
        onChange={doMap}
      />
      {value && (
        <div className={styles.info} data-testid='info'>
          <Text color='tertiary' size='small'>
            Roman notation:
          </Text>
          <Text color='tertiary' size='large'>
            {value}
          </Text>
        </div>
      )}
      {error && (
        <div className={styles.error} data-testid='error'>
          <Text color='danger'>{error}</Text>
        </div>
      )}
    </Card>
  );
};
