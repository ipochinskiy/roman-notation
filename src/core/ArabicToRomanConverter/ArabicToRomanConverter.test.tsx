import React from 'react';
import { shallow } from 'enzyme';

import { ArabicToRomanConverter } from './ArabicToRomanConverter';

describe('ArabicToRomanConverter', () => {
  it('should render correctly', () => {
    const component = shallow(<ArabicToRomanConverter />);

    expect(component).toMatchSnapshot();
  });
});
