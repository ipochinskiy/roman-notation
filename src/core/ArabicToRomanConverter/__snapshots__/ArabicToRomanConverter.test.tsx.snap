// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArabicToRomanConverter should render correctly 1`] = `
<div
  className="converter"
>
  <Card>
    <Input
      fullWidth={true}
      onChange={[Function]}
      placeholder="42"
      type="number"
      value="42"
    />
    <div
      className="info"
    >
      <Text
        color="tertiary"
        size="small"
      >
        Roman notation:
      </Text>
      <Text
        color="tertiary"
        size="large"
      >
        42
      </Text>
    </div>
  </Card>
</div>
`;
